core/src/main/scala/org/bitcoins/core/wallet/fee/FeeUnit.scala

```diff
@@ -25,17 +25,23 @@ case class SatoshisPerByte(currencyUnit: CurrencyUnit) extends BitcoinFeeUnit {
   def toSatPerKb: SatoshisPerKiloByte = {
     SatoshisPerKiloByte(currencyUnit.satoshis * Satoshis(1000))
   }
+
+  override def calc(tx: Transaction): CurrencyUnit =
+    Satoshis(tx.byteSize * toLong)
 }
 
 object SatoshisPerByte {
   def fromLong(sats: Long): SatoshisPerByte = SatoshisPerByte(Satoshis(sats))
 }
 
+/**
+  * KiloBytes here are defined as 1000 bytes.
+  */
 case class SatoshisPerKiloByte(currencyUnit: CurrencyUnit)
     extends BitcoinFeeUnit {
 
   def toSatPerByte: SatoshisPerByte = {
-    val conversionOpt = (currencyUnit.toBigDecimal * 0.001).toBigIntExact
+    val conversionOpt = (currencyUnit.toBigDecimal / 1000).toBigIntExact
```

Comment:
Is there precision issues here?

Reply:
I think using `Exact` means it throws if there were to be any precision issue?